lnbits/core/crud.py

```diff
@@ -620,3 +620,40 @@ async def create_admin_settings(super_user: str, new_settings: dict):
     sql = f"INSERT INTO settings (super_user, editable_settings) VALUES (?, ?)"
     await db.execute(sql, (super_user, json.dumps(new_settings)))
     return await get_super_settings()
+
+
+# tinyurl
+# -------
+
+
+async def create_tinyurl(domain: str, endless: bool, conn: Optional[Connection] = None):
+    tinyurl_id = uuid4().hex[:8]
```

Comment:
- `uuid4().hex` returns the `hex` (base16) value. This means only characters in the range `0-1` and `a-f`

- `[:8]` only the first 8 characters (eg: `1ede84bc`)
  - one character represents 2 `bytes`
  - this means the max value range is `2^16` (`65,536`). This is not such a big number, collisions might occur

### Suggestion
This already exists:
```py
def urlsafe_short_hash() -> str:
    return shortuuid.uuid()
```

[shortuuid doc:](https://pypi.org/project/shortuuid/)
> shortuuid solves this problem by generating uuids using Python's built-in uuid module and then translating them to **base57** using lowercase and uppercase letters and digits, and removing similar-looking characters such as l, 1, I, O and 0.

- the **base57** encoding from `shortuuid` would significantly increase the values space

Reply:
Great explanation. Thanks! Done